delphi/translators/for2py/rectify.py

```diff
@@ -630,39 +1079,44 @@ def handle_tag_value(self, root, parElem):
         """
         for child in root:
             self.clean_attrib(child)
-            curElem = ET.SubElement(parElem, child.tag, child.attrib)
-            if (
-                child.tag == "literal"
-                or child.tag == "operation"
-                or child.tag == "name"
-            ):
-                self.parseXMLTree(child, curElem)
-                if child.tag == "name" and self.need_reconstruct:
-                    self.reconstruct_name_element(curElem, parElem)
-            else:
-                print(f'In self.handle_tag_value: "{child.tag}" not handled')
+            cur_elem = ET.SubElement(current, child.tag, child.attrib)
+
+            try:
+                error_chk = self.value_child_tags.index(child.tag)
+            except ValueError:
+                assert False, f'In handle_tag_value: "{child.tag}" not handled'
+
+            self.parseXMLTree(child, cur_elem, current, parent, traverse)
+            if child.tag == "name" and self.need_reconstruct:
+                self.reconstruct_name_element(cur_elem, current)
 
-    def handle_tag_literal(self, root, parElem):
+    def handle_tag_literal(self, root, current, parent, grandparent, traverse):
         """
             This function handles cleaning up the XML elementss between the literal elementss.
             <literal>
                 ...
             </literal>
         """
-        if '"' in parElem.attrib["value"]:
-            parElem.attrib["value"] = self.clean_id(parElem.attrib["value"])
+        if '"' in current.attrib['value']:
+            current.attrib['value'] = self.clean_id(current.attrib['value'])
         for child in root:
             self.clean_attrib(child)
```

Comment:
Lines 1103-1117 seem very similar (though not always identical) to lines 1127-1141 and 1152-1165.  Is there anything we can do to reduce or eliminate the duplicated code?